crates/ruff_db/src/panic.rs

```diff
@@ -1,20 +1,27 @@
 use std::cell::Cell;
+use std::panic::Location;
 use std::sync::OnceLock;
 
 #[derive(Default, Debug)]
 pub struct PanicError {
-    pub info: String,
+    pub location: Option<String>,
+    pub payload: Option<String>,
```

Comment:
Note that we can't put the actual payload here, because what we get back is a _reference_ to a `dyn Any`, and that does not guarantee that the payload is cloneable.